martian/src/metadata.rs

```diff
@@ -170,13 +198,15 @@ impl<'a> Metadata<'a> {
 
     /// Write finalized _jobinfo data
     pub fn update_jobinfo(&mut self) -> Result<()> {
-        let mut jobinfo = self.read_json_obj("jobinfo")?;
+        let mut raw_jobinfo = self.read_json_obj("jobinfo")?;
+        let jobinfo: JobInfo = serde_json::from_value(Value::Object(raw_jobinfo.clone()))?;
 
         let exe = env::current_exe()?.to_str().unwrap().to_string();
-        jobinfo.insert("rust_exe".to_string(), Value::String(exe));
-        // jobinfo.insert("rust_version", sys.version);
+        raw_jobinfo.insert("rust_exe".to_string(), Value::String(exe));
+        // raw_jobinfo.insert("rust_version", sys.version);
```

Comment:
To be more consistent with the way python reports this stuff, it would be better to have a "rust" key with a dictionary containing both the rustc version and the executable path.

Reply:
This is what I see in a jobinfo file:
```
"python": {
    "binpath": "/path/to/python",
    "version": "2.7.14 |Anaconda, Inc.| (default, Mar 27 2018, 17:29:31) \n[GCC 7.2.0]",
}
```
Shall I stick to the same convention and use the same keys for `rust` too?

Reply:
Yes that would be reasonable.  The golang adapter should probably likewise be updated to do that, but then again no one actually uses it...